packages/cli-kit/src/private/node/ui/hooks/use-async.ts

```diff
@@ -0,0 +1,26 @@
+import {useApp} from 'ink'
+import {useEffect} from 'react'
+
+interface Options {
+  onResolve?: () => unknown
+  onReject?: (error: Error) => void
+}
+
+export default function useAsync(
+  asyncFunction: () => Promise<unknown>,
+  {onResolve = () => {}, onReject = () => {}}: Options,
+) {
+  const {exit: unmountInk} = useApp()
+
+  useEffect(() => {
+    asyncFunction()
+      .then(() => {
+        onResolve()
+        unmountInk()
```

Comment:
What does `unmounting` mean in the context of a component?

Reply:
Also, that side effect doesn't seem well captured in the name of the function, `useAsync`.

Reply:
Basically, this hook allows you to wait inside react components for async behavior. Once this behavior resolves we want to unmount the rendering `ink` instance explicitly so that the promise we're awaiting from the outside (`waitForExit`) resolves.

Regarding the naming I'm open to suggestions. Would something like `awaitAndUnmount` be better?

Reply:
@pepicrft how about `useEffectAsync`?